connectivity/FEATURE_BLE/source/generic/SecurityManagerImpl.cpp

```diff
@@ -1199,11 +1445,13 @@ void SecurityManager::enable_encryption_cb(
     MBED_ASSERT(_db);
     ControlBlock_t *cb = get_control_block(db_entry);
     if (!cb) {
+        tr_error("Failure, control block %p not available", db_entry);
         return;
     }
 
     SecurityDistributionFlags_t* flags = _db->get_distribution_flags(cb->db_entry);
     if (!flags) {
+        tr_error("Connection %d - Cannot enable encryption, no distribution flags found for entry %p", cb->connection, cb->db_entry);
```

Comment:
honestly, if you're fixing things, this whole if statement should just be an `MBED_ASSERT(flags);`

Reply:
Same below

Reply:
The fix would be to validate the DB when it is loaded. I'm not doing it now.